airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -2172,6 +2176,16 @@ def _build_stream_slicer_from_partition_router(
                     ],
                     parameters={},
                 )
+            elif isinstance(stream_slicer_model, dict):
+                # partition router comes from CustomRetrieverModel and has not been and therefore partition_router has not been parsed as a model
```

Comment:
nit: i think you meant to say "partition router comes from CustomRetrieverModel and has therefore has not been parsed as a model"